### PR TITLE
[SPARK-42424][YARN] Remove unused declarations  from Yarn module

### PR DESCRIPTION
**Pull request by LuciferYang**
### What changes were proposed in this pull request?
This pr cleans up unused declarations in the Yarn module:

- `YarnSparkHadoopUtil#setEnvFromInputString`: this method Introduced by SPARK-3477 and becomes a unused and no test coverage method after SPARK-17979

- `YarnSparkHadoopUtil#environmentVariableRegex`: this `val` is only used by `YarnSparkHadoopUtil#setEnvFromInputString`

- `ApplicationMasterArguments.DEFAULT_NUMBER_EXECUTORS`: this `val` Introduced by SPARK-1946 and replaced by `YarnSparkHadoopUtil.DEFAULT_NUMBER_EXECUTORS` in SPARK-4138, and It was eventually replaced by `config#EXECUTOR_INSTANCES`

- `ApplicationMaster.EXIT_SECURITY`: this `val` introduced by SPARK-3627 and it is used to represent the exit code of errors related to `System.setSecurityManager`, SPARK-4584 deleted the use of `SecurityManager` and this val is useless.

### Why are the changes needed?
Code clean up.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Pass GitHub Actions

### CODE REVIEW

**Comment by LuciferYang** [comment]
cc @HyukjinKwon @srowen @dongjoon-hyun @tgravescs  @mridulm

**Review by srowen — COMMENTED** [review]
Sounds fine, is there more unused code we can just prune all at once?

**Review by dongjoon-hyun — COMMENTED** [review]
+1 for Sean's comment.

**Comment by LuciferYang** [comment]
> Sounds fine, is there more unused code we can just prune all at once?

I think there should be, in what scope? Yarn module scope or Spark project scope?

**Comment by srowen** [comment]
Maybe just YARN to start

**Comment by LuciferYang** [comment]
OK

**Comment by LuciferYang** [comment]
[92f9b38](https://github.com/apache/spark/pull/39997/commits/92f9b38d0641d3db7c751e9d155fbacb3df776a2) include used local vars, please let me know if you don't need to change them.

**Comment by LuciferYang** [comment]
> Maybe just YARN to start

Should be no more cases in Yarn source code (this pr not include test code now)

File: resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala [hunk 1]
```diff
@@ -595,7 +595,7 @@ private[spark] class ApplicationMaster(
         }
         failureCount = 0
       } catch {
-        case i: InterruptedException => // do nothing
+        case _: InterruptedException => // do nothing
```
Comment (srowen) [inline resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala:598]: These are not important, but OK if we're just doing cleanup. Seems OK

Comment (LuciferYang) [reply resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala:598]: Yeah, if it's not important, I can roll back these. They are in a commit

Comment (LuciferYang) [reply resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala]: @srowen [53a6bfa](https://github.com/apache/spark/pull/39997/commits/53a6bfa6373872e3a8edaa531486d82e593d0ea7) revert [92f9b38](https://github.com/apache/spark/pull/39997/commits/92f9b38d0641d3db7c751e9d155fbacb3df776a2) and update pr description

**Review by srowen — APPROVED** [review]
OK pending tests

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM from my side.

cc @mridulm and @tgravescs once more.

**Review by mridulm — APPROVED** [review]

**Comment by srowen** [comment]
Merged to master

**Pull request closed without merging**